photon_stream/io/jsonl.py

```diff
@@ -116,10 +116,13 @@ def __init__(self, fin):
     def __iter__(self):
         return self
 
+    def __readline(self):
+        return self.fin.__next__().strip().rstrip(',') # raises StopIteration
```

Comment:
So here the StopIterationException is raised at End-of-File.

Reply:
Exactly. Whenever the underlying stream raises a `StopIteration`, so will the JSONLines reader.